test_xdb/models/under_test/cast_json_test.sql

```diff
@@ -0,0 +1,20 @@
+{{ config({"tags":["exclude_bigquery", "exclude_bigquery_tests"]}) }}
+
+with test_raw_data as (
+    select
+        '{"key1":"value1", "key2":{"innerkey2_1":"innervalue2_1", "0":[0, null, "string"]}, "key3": -1234}'
+            as varchar_json
+)
+, test_json_data as (
+    select {{xdb.cast_json("varchar_json")}} as json_col from test_raw_data
+)
+
+select
+    {% if target.type == 'postgres' %}
+        json_col->>'key1' as value1,
```

Comment:
does it make sense to add another helper macro that outputs either `->>` or `:` for accessing the json value?

Reply:
I actually avoided creating one that uses those accessors in favor of using the `json_extract_path_text` function in the identically named macro [here](https://github.com/Health-Union/dbt-xdb/blob/master/macros/json_extract_path_text.sql), which should mostly accomplish the same thing. I also think it  circumvents the slight variations in datatypes returned by certain accessors that might require post-processing to have more uniform behavior between databases.